src/annotator/guest.js

```diff
@@ -439,10 +432,16 @@ export default class Guest {
       annotations => annotations.forEach(annotation => this.anchor(annotation))
     );
 
-    // Discover and connect to the sidebar frame. All RPC events must be
-    // registered before creating the channel.
-    const sidebarPort = await this._portFinder.discover('sidebar');
-    this._sidebarRPC.connect(sidebarPort);
+    // Connect to sidebar and send document info/URIs to it.
+    //
+    // RPC calls are deferred until a connection is made, so these steps can
+    // complete in either order.
+    this._portFinder.discover('sidebar').then(port => {
+      this._sidebarRPC.connect(port);
+    });
+    this.getDocumentInfo().then(metadata =>
+      this._sidebarRPC.call('documentInfoChanged', metadata)
+    );
```

Comment:
If getting document info fails, the sidebar will remain in a loading state indefinitely. This was the case before as well. To improve this we'd need to add an appropriate error UI in the sidebar and also have a way for the guest to notify the sidebar about the error.

Reply:
> Alternative, we could continue to send an error and data (following the Node.js convention).

You mean make the arguments to `documentInfoChanged` have the form `(err, data)`? That could work. Another option would be to use a variant/union.